### PR TITLE
258 : Add Lookup for clients using Client Type and Location Id

### PR DESCRIPTION
**Pull request by rehammuzzamil**
Issue # 258
Dependent on configs PR : https://github.com/OpenSRP/opensrp-server-configs/pull/78

### CODE REVIEW

File: src/main/java/org/opensrp/repository/postgres/ClientsRepositoryImpl.java [hunk 1]
```diff
@@ -661,6 +663,13 @@ public void removeAll() {
 	public List<Patient> findClientByRelationship(String relationship, String id) {
 		return convertToFHIR(findByRelationshipId(relationship, id));
 	}
+
+	@Override
+	public List<Client> findByClientTypeAndLocationId(String clientType, String locationId) {
```
Comment (githengi) [inline src/main/java/org/opensrp/repository/postgres/ClientsRepositoryImpl.java:668]: change `findFamilyMemberyByJurisdiction` to use this method

Comment (githengi) [reply src/main/java/org/opensrp/repository/postgres/ClientsRepositoryImpl.java:668]: change `findFamilyMemberByResidence`  this clause ` .andLastNameNotEqualTo("Family");` to ` .andClientTypeNotEqualTo("Family");`

File: src/test/java/org/opensrp/repository/postgres/ClientsRepositoryTest.java [hunk 2]
```diff
@@ -697,5 +697,15 @@ public void testGetAllIdsShouldGetAllArchivedClientIds() {
...
 		assertEquals(1573733955111l, idsModel.getRight().longValue());
 	}
+
+	@Test
+	public void testFindByClientTypeAndLocationId() {
+		Client client = new Client("f67823b0-378e-4a35-93fc-bb00def74e24").withLocationId("location-1");
+		client.setClientType("Client-type-1");
+		clientsRepository.add(client);
+		List<Client> clients = clientsRepository.findByClientTypeAndLocationId("Client-type-1","location-1");
+		assertEquals(1,clients.size());
```
Comment (githengi) [inline src/test/java/org/opensrp/repository/postgres/ClientsRepositoryTest.java:707]: Assert the client id at the least is as expected

Comment (rehammuzzamil) [reply src/test/java/org/opensrp/repository/postgres/ClientsRepositoryTest.java:707]: > Assert the client id at the least is as expected

Added assertion on base entity id

File: src/main/java/org/opensrp/repository/postgres/ClientsRepositoryImpl.java (hunk 1 above)
Comment (rehammuzzamil) [reply src/main/java/org/opensrp/repository/postgres/ClientsRepositoryImpl.java:663]: Done

Comment (rehammuzzamil) [reply src/main/java/org/opensrp/repository/postgres/ClientsRepositoryImpl.java:663]: Done

**Pull request merged**